product.go

```diff
@@ -152,3 +165,23 @@ func (s *ProductServiceOp) DeleteMetafield(productID int64, metafieldID int64) e
 	metafieldService := &MetafieldServiceOp{client: s.client, resource: productsResourceName, resourceID: productID}
 	return metafieldService.Delete(metafieldID)
 }
+
+// Publish or UnPublish  an existing product
+func (s *ProductServiceOp) PublishOrUnPublish(productID int64, isPublish bool) (*Product, error) {
+	path := fmt.Sprintf("%s/%s/%d.json", globalApiPathPrefix, productsBasePath, productID)
+	if isPublish {
+		now := time.Now()
```

Comment:
I think it will be better to use `time.Now().UTC()` here
Also according to Shopify documentation they have the property "published" = true/false and you can use it to publish/unpublish the request
https://shopify.dev/docs/admin-api/rest/reference/products/product#Show%20a%20hidden%20product%20by%20changing%20the%20published%20attribute%20to%20true

I would suggest to research it a bit and if the API allows you to set `published_at` to the future date, pass the date as a parameter. If not - just use the `published` property instead